src/Diff.re

```diff
@@ -3,7 +3,7 @@ let maxYIQPossibleDelta = 35215.;
 
 type diffVariant('a) =
   | Layout
-  | Pixel(('a, int, float));
+  | Pixel(('a, int, float, Option.t(Stack.t(int))));
```

Comment:
This could be just `option` instead of `Option.t`. But I don't think it makes a difference.

I don't know however if the option is needed at all. I think it allocates more memory putting the stack inside the option instead of just creating an empty stack directly.

Then you could also collapse the switch statement later to a simple boolean if condition checking the `diffLines` flag.

Reply:
Thanks for the review! I forgot that ocaml options are boxed :)